### PR TITLE
Don't just put the errors, catch them.

### PR DESCRIPTION
**Pull request by astuyve**
Best way to test is to pull the branch, bundle against it, and try to update a subscription.

### CODE REVIEW

**Comment by astuyve** [comment]
@jonkarna QACR?

File: lib/zuora_client.rb [hunk 1]
```diff
@@ -122,51 +122,46 @@ def query(query_string)
 
     def subscribe(obj)
       begin
-        response = @client.api_call(:subscribe, obj)
-        return response
+        @client.api_call(:subscribe, obj)
       rescue Exception => e
-        puts e.message
+        [errors: {message: e.message, code: nil}, object: obj]
```
Comment (jonkarna) [inline lib/zuora_client.rb]: Since this doesn't do the same hashing, we can't output something similar to what would normally be output.

Comment (astuyve) [inline lib/zuora_client.rb]: What should we output then? Just the message alone, or in an array of some kind?

File: lib/zuora_client.rb [hunk 2]
```diff
@@ -132,41 +131,37 @@ def subscribe(obj)
     def create(obj)
       begin
         response = @client.api_call(:create,obj)
-        result = save_results_to_hash(response)
+        save_results_to_hash(response)
       rescue Exception => e
-        puts e.message
+        [errors: {message: e.message, code: nil}, object: obj]
```
Comment (jonkarna) [inline lib/zuora_client.rb]: Errors is suppose to be an array of hashes

```
[errors: [{ message: e.message, code: ni l}], object: obj]
```

**Comment by jonkarna** [comment]
:+1: QACR

**Pull request merged**